File: cafeoccidente-backend/src/main/java/com/cafeoccidente/backend/purchases/husk/service/HuskPurchaseService.java
package com.cafeoccidente.backend.purchases.husk.service;

import com.cafeoccidente.backend.purchases.husk.dto.AnnouncementInfoResponse;
import com.cafeoccidente.backend.purchases.husk.dto.HuskPurchaseRequest;
import com.cafeoccidente.backend.purchases.husk.dto.HuskPurchaseResponse;
import com.cafeoccidente.backend.purchases.husk.dto.NextInvoiceNumberResponse;

public interface HuskPurchaseService {
    HuskPurchaseResponse create(HuskPurchaseRequest request);

    HuskPurchaseResponse findById(Long id);

    HuskPurchaseCalculation preview(HuskPurchaseRequest request);

    /** Factura: siguiente consecutivo propio de PASILLA. */
    NextInvoiceNumberResponse nextInvoiceNumber();

    /** Cod Prod + anuncio vigente para la agencia del usuario (Fondo "RP" / Especial "PASILLA"). */
    AnnouncementInfoResponse announcementInfo();
}

File: cafe-occidente-frontend/src/app/core/services/husk-purchase.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';

import { API_BASE_URL } from '../config/api.config';
import {
  HuskAnnouncementInfo,
  HuskNextInvoiceNumber,
  HuskPurchaseCalculation,
  HuskPurchaseRequest,
  HuskPurchaseResponse,
} from '../models/husk-purchase.model';

/** Unica responsabilidad: llamadas HTTP para el formulario Compra Pasilla (PASILLA). */
@Injectable({ providedIn: 'root' })
export class HuskPurchaseService {
  private readonly http = inject(HttpClient);

  create(request: HuskPurchaseRequest): Observable<HuskPurchaseResponse> {
    return this.http.post<HuskPurchaseResponse>(`${API_BASE_URL}/purchases/husk`, request);
  }

  preview(request: HuskPurchaseRequest): Observable<HuskPurchaseCalculation> {
    return this.http.post<HuskPurchaseCalculation>(`${API_BASE_URL}/purchases/husk/preview`, request);
  }

  /** Factura: siguiente consecutivo propio de PASILLA. */
  nextInvoiceNumber(): Observable<HuskNextInvoiceNumber> {
    return this.http.get<HuskNextInvoiceNumber>(`${API_BASE_URL}/purchases/husk/next-invoice-number`);
  }

  /** Cod Prod + anuncio vigente (Especial "PASILLA" / Fondo "RP" fijos). */
  announcementInfo(): Observable<HuskAnnouncementInfo> {
    return this.http.get<HuskAnnouncementInfo>(`${API_BASE_URL}/purchases/husk/announcement-info`);
  }
}
